transfer-manager/contracts/RaribleTransferManager.sol

```diff
@@ -60,39 +69,42 @@ abstract contract RaribleTransferManager is OwnableUpgradeable, ITransferManager
     }
 
     function doTransfers(
-        LibAsset.AssetType memory makeMatch,
-        LibAsset.AssetType memory takeMatch,
-        LibFill.FillResult memory fill,
-        LibOrder.Order memory leftOrder,
-        LibOrder.Order memory rightOrder,
-        LibOrderDataV2.DataV2 memory leftOrderData,
-        LibOrderDataV2.DataV2 memory rightOrderData
-    ) override internal returns (uint totalMakeValue, uint totalTakeValue) {
-        LibFeeSide.FeeSide feeSide = LibFeeSide.getFeeSide(makeMatch.assetClass, takeMatch.assetClass);
-        totalMakeValue = fill.leftValue;
-        totalTakeValue = fill.rightValue;
+        LibAsset.Asset memory makeMatch,
+        LibAsset.Asset memory takeMatch,
+        LibDeal.Data memory left,
+        LibDeal.Data memory right,
+        address leftMaker,
+        address rightMaker,
+        address originalMessageSender
+    ) override payable external {
+        LibFeeSide.FeeSide feeSide = LibFeeSide.getFeeSide(makeMatch.assetType.assetClass, takeMatch.assetType.assetClass);
         if (feeSide == LibFeeSide.FeeSide.MAKE) {
-            totalMakeValue = doTransfersWithFees(fill.leftValue, leftOrder.maker, leftOrderData, rightOrderData, makeMatch, takeMatch,  TO_TAKER);
-            transferPayouts(takeMatch, fill.rightValue, rightOrder.maker, leftOrderData.payouts, TO_MAKER);
+            doTransfersWithFees(makeMatch.value, leftMaker, left, right, makeMatch.assetType, takeMatch.assetType, TO_TAKER);
+            transferPayouts(takeMatch.assetType, takeMatch.value, rightMaker, left.payouts, TO_MAKER);
         } else if (feeSide == LibFeeSide.FeeSide.TAKE) {
-            totalTakeValue = doTransfersWithFees(fill.rightValue, rightOrder.maker, rightOrderData, leftOrderData, takeMatch, makeMatch, TO_MAKER);
-            transferPayouts(makeMatch, fill.leftValue, leftOrder.maker, rightOrderData.payouts, TO_TAKER);
+            doTransfersWithFees(takeMatch.value, rightMaker, right, left, takeMatch.assetType, makeMatch.assetType, TO_MAKER);
+            transferPayouts(makeMatch.assetType, makeMatch.value, leftMaker, right.payouts, TO_TAKER);
         } else {
-            transferPayouts(makeMatch, fill.leftValue, leftOrder.maker, rightOrderData.payouts, TO_TAKER);
-            transferPayouts(takeMatch, fill.rightValue, rightOrder.maker, leftOrderData.payouts, TO_MAKER);
+            transferPayouts(makeMatch.assetType, makeMatch.value, leftMaker, right.payouts, TO_TAKER);
+            transferPayouts(takeMatch.assetType, takeMatch.value, rightMaker, left.payouts, TO_MAKER);
+        }
+        /*Return back eth, if eth more than need*/
+        uint ethBalance = address(this).balance;
+        if (ethBalance > 0) {
+            address(originalMessageSender).transferEth(ethBalance);
```

Comment:
not sure if transferring all balance is ok

Reply:
eah, send eth in call doTransfers()
inside  doTransfers() get eth by msg.value